core/src/main/java/org/apache/iceberg/encryption/StandardEncryptionManager.java

```diff
@@ -51,6 +69,8 @@ public StandardEncryptionManager(
     this.tableKeyId = tableKeyId;
     this.kmsClient = kmsClient;
     this.dataKeyLength = dataKeyLength;
+    this.kekCacheTimeout = kekCacheTimeout;
+    this.kekCache = Maps.newHashMap();
```

Comment:
Should this use a `Cache` that has a time-based eviction policy? (See the [Caffeine javadocs](https://www.javadoc.io/doc/com.github.ben-manes.caffeine/caffeine/2.1.0/com/github/benmanes/caffeine/cache/Caffeine.html))

```java
   LoadingCache<String, ByteBuffer> kekCache = Caffeine.newBuilder()
       .expireAfterWrite(kekCacheTimeout, TimeUnit.MINUTES)
       .build(key -> unwrapKey(key));
```

Reply:
Probably `kekCacheTimeout` is not an accurate name - the parameter basically means how long we can use a KEK in the writer, before we need to generate a new one. I'll rename it.
We need previous keys in the cache, so that readers don't call the KMS again.